### PR TITLE
Use correct free plan in canUpgradeToPlan selector

### PR DESCRIPTION
**Pull request by sirreal**
@rickybanister reported missing purchase buttons on his plans page. I've localized the problem but I'm not clear on steps to reproduce. This PR fixing the problem.

The only usage of the modified selector I see in the codebase is the affected one, so it's unlikely other behavior is modified.

The selector uses a utility function to test whether a plan can be purchase when another plan is currently owned and active. It tests the currently owned plan.

Within the selector, if the currently owned plan is expired, it replaces the currently owned plan with a free plan. This was hard-coded as a free WordPress.com plan rather than a Jetpack plan, which resulted in no other Jetpack plans being purchaseable.

This PR changes the logic to verify whether the site is Jetpack and not Atomic, and in that case uses the Jetpack Free plan which corrects the issue.

## Testing

As I said, I haven't been able to reproduce, but if you have a Jetpack site with an expired plan you should not be able to purchase a plan from `/plans/:SITE`. The buttons are hidden (see before). With this PR the buttons are displayed correctly when you own an expired Jetpack plan.

Be sure to verify that other plans pages continue to behave correctly!
* Atomic sites
  * free
  * paid
  * expired
* WordPress.com sites
  * free
  * paid
  * expired
* Jetpack sites
  * free
  * paid
  * expired

### Before

before

### After

after

### CODE REVIEW

**Comment by rickybanister** [comment]
Thanks for the very very quick address of the issue @sirreal. I think your solution (letting the user re-purchase whatever they want) makes sense for now, but I did do another mockup of displaying the plans page more in an expired state than the default state:

screen shot 2017-08-18 at 11 14 43 am

File: client/lib/plans/index.js [hunk 1]
```diff
@@ -62,15 +54,26 @@ export function getFeatureTitle( feature ) {
...
 export function canUpgradeToPlan( planKey, site ) {
-	const plan = get( site, [ 'plan', 'expired' ], false ) ? PLAN_FREE : get( site, [ 'plan', 'product_slug' ], PLAN_FREE );
+	// Which "free plan" should we use to test
+	const freePlan =
+		get( site, 'jetpack', false ) && ! get( site, [ 'options', 'is_automated_transfer' ], false )
+			? PLAN_JETPACK_FREE
+			: PLAN_FREE;
+	console.log( 'jp', get( site, 'jetpack', false ) );
+	console.log( 'at', get( site, [ 'options', 'is_automated_transfer' ], false ) );
+	console.log( 'plan', freePlan );
```
Comment (oskosk) [inline client/lib/plans/index.js]: 👀

Comment (sirreal) [inline client/lib/plans/index.js]: Sorry, see #17039 😩 . Just fixed this.

**Comment by sirreal** [comment]
Added tests that would have uncovered the issue.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+	PLAN_JETPACK_PERSONAL_MONTHLY,
+	PLAN_JETPACK_PREMIUM,
+	PLAN_JETPACK_PREMIUM_MONTHLY,
+	PLAN_PERSONAL,
+	PLAN_PREMIUM,
+} from '../constants';
+
+describe( 'canUpgradeToPlan', () => {
+	it( 'should return true from lower-tier plans to higher-tier plans', () => {
+		const makeSite = product_slug => ( {
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Let's move the `makeSite` utility in the main `describe` call, so we don't have to repeat it in every test. You can then create a different one for the last test.

Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Also, while I understand naming the argument `product_slug` makes the code shorter, it's not compliant to our coding standards, so we should probably consider renaming it to `productSlug`.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_FREE, PLAN_PERSONAL ],
+			[ PLAN_PERSONAL, PLAN_PREMIUM ],
+			[ PLAN_PREMIUM, PLAN_BUSINESS ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: We're also able to upgrade:

* From a Free plan to Premum plan.
* From a Free plan to Business plan.
* From a Personal plan to a Business plan.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			[ PLAN_JETPACK_FREE, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_PERSONAL, PLAN_JETPACK_PREMIUM ],
+			[ PLAN_JETPACK_PREMIUM, PLAN_JETPACK_BUSINESS ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Jetpack monthly plans are missing here: user is able to upgrade:

* From `PLAN_JETPACK_FREE` to any monthly Jetpack plan.
* From any monthly Jetpack plan to a higher monthly Jetpack plan.
* From any lower yearly plan to a higher monthly plan.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+	it( 'should return true from monthly plans to yearly plans', () => {
+		const makeSite = product_slug => ( {
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_JETPACK_PERSONAL_MONTHLY, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_PREMIUM_MONTHLY, PLAN_JETPACK_PREMIUM ],
+			[ PLAN_JETPACK_BUSINESS_MONTHLY, PLAN_JETPACK_BUSINESS ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Users are also able to upgrade from lower Jetpack monthly plans to higher yearly plans:
* Personal monthly to Premium yearly
* Personal monthly to Professional yearly
* Premium monthly to Professional yearly

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+	it( 'should return true from lower-tier plans to higher-tier plans', () => {
+		const makeSite = product_slug => ( {
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_FREE, PLAN_PERSONAL ],
+			[ PLAN_PERSONAL, PLAN_PREMIUM ],
+			[ PLAN_PREMIUM, PLAN_BUSINESS ],
+			[ PLAN_JETPACK_FREE, PLAN_JETPACK_PERSONAL ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Users are also able to upgrade from Jetpack free to any yearly Jetpack plan, not just Personal.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+			[ PLAN_JETPACK_PERSONAL, PLAN_JETPACK_PREMIUM ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Let's make sure to not skip any other possibilities, like Personal to Professional for example.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+		);
+	} );
+
+	it( 'should return false from yearly plans to monthly plans', () => {
+		const makeSite = product_slug => ( {
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_JETPACK_PERSONAL, PLAN_JETPACK_PERSONAL_MONTHLY ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: As mentioned above, this is not true for the case when upgrading from a lower tier yearly plan to a higher tier monthly plan. We should probably add a new test with those cases.

File: client/lib/plans/test/can-upgrade-to-plan.js [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+			plan: { product_slug },
+		} );
+
+		[
+			[ PLAN_PERSONAL, PLAN_FREE ],
+			[ PLAN_PREMIUM, PLAN_PERSONAL ],
+			[ PLAN_BUSINESS, PLAN_PREMIUM ],
+			[ PLAN_JETPACK_PERSONAL, PLAN_JETPACK_FREE ],
+			[ PLAN_JETPACK_PREMIUM, PLAN_JETPACK_PERSONAL ],
+			[ PLAN_JETPACK_BUSINESS, PLAN_JETPACK_PREMIUM ],
```
Comment (tyxla) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Let's include all other possibilities here, including the combinations with Jetpack monthly plans.

File: client/lib/plans/test/can-upgrade-to-plan.js (hunk 2 above)
Comment (sirreal) [inline client/lib/plans/test/can-upgrade-to-plan.js]: Done and done 🙂 
7393e6c
f787b93

File: client/lib/plans/test/can-upgrade-to-plan.js (hunk 3 above)
Comment (sirreal) [inline client/lib/plans/test/can-upgrade-to-plan.js]: 👍  I made this test quite exhaustive in 0d1f6d2

File: client/lib/plans/test/can-upgrade-to-plan.js (hunk 5 above)
Comment (sirreal) [inline client/lib/plans/test/can-upgrade-to-plan.js]: 👌  Should be covered by the lower to higher tier tests in 0d1f6d2

**Comment by sirreal** [comment]
@tyxla Thanks for the feedback, I've made the tests much more exhaustive. Care to have another look?

**Pull request merged**